### PR TITLE
fix(header/p2p): set deadline on a request to peer

### PR DESCRIPTION
**Pull request by Wondertan**

### CODE REVIEW

File: header/p2p/session.go [hunk 1]
```diff
@@ -130,14 +131,13 @@ func (s *session) doRequest(
...
+	defer cancel()
+
 	r, size, duration, err := sendMessage(ctx, s.host, stat.peerID, s.protocolID, req)
 	if err != nil {
-		if err == context.Canceled || err == context.DeadlineExceeded {
-			return
-		}
 		log.Errorw("requesting headers from peer failed.", "failed peer", stat.peerID, "err", err)
 		select {
-		case <-ctx.Done():
```
Comment (vgonkivs) [inline header/p2p/session.go:140]: why? we will request headers from another peer. Previous one, will not be added

Comment (Wondertan) [reply header/p2p/session.go:140]: because then request won't be retried!

Comment (Wondertan) [reply header/p2p/session.go:140]: ctx.Done succeds and the `case s.reqCh <- req` does not do anything

Comment (vgonkivs) [reply header/p2p/session.go:140]: yes, correctly. Sorry, I was blind on Friday

**Comment by Wondertan** [comment]
Closing in favour of https://github.com/celestiaorg/celestia-node/pull/1603

**Pull request closed without merging**